app/controllers/api/v2/courses_controller.rb

```diff
@@ -25,6 +25,14 @@ def sync_with_search_and_compare
         head response ? :ok : :internal_server_error
       end
 
+      def publish
+        if @provider.opted_in
+          @course.publish_sites
+        end
+        @course.publish_enrichment(@current_user.id)
+        sync_with_search_and_compare # todo: make a service?
```

Comment:
Definitely should not be calling an action, and should probably be a service.

Reply:
As agreed in person, for this PR how about we just duplicate what's in `sync_with_search_and_compare` here and we can (maybe) turn it into a service later.

Reply:
done